src/GraphQL/ResolveFieldContextExtensions.cs

```diff
@@ -58,15 +58,17 @@ private static bool TryGetArgument(this IResolveFieldContext context, System.Typ
         /// <summary>Determines if the specified field argument has been provided in the GraphQL query request</summary>
         public static bool HasArgument(this IResolveFieldContext context, string argumentName) => context.Arguments?.ContainsKey(argumentName) ?? false;
 
-        internal static IResolveFieldContext<TSourceType> As<TSourceType>(this IResolveFieldContext context)
+        /// <summary>Returns the <see cref="IResolveFieldContext"/> typed as an <see cref="IResolveFieldContext{TSource}"/></summary>
```

Comment:
It would be good to mention throw conditions (inside ResolveFieldContextAdapter).

Reply:
Done; added to `As`, `ResolveFieldContextAdapter`, and the constructor of `ResolveFieldContext<TSource>`